homeassistant/components/zwave/__init__.py

```diff
@@ -376,6 +376,26 @@ def _on_timeout(sec):
         hass.add_job(check_has_unique_id, entity, _on_ready, _on_timeout,
                      hass.loop)
 
+    def node_removed(node):
+        node_id = node.node_id
+        node_key = 'node-{}'.format(node_id)
+        _LOGGER.info("Node Removed: %s",
+                     hass.data[DATA_DEVICES][node_key])
+        for key in list(hass.data[DATA_DEVICES]):
+            if not key.startswith('{}-'.format(node_id)):
```

Comment:
If the node key is formatted as `node-<node id>`, why check if the key starts with node id ?

Reply:
The keys are the same as the unique_id that gets stored in the entity registry. For the `zwave.device_xyz` entities the unique ID is `node-nodeID` or(ex:  `node-12`) - for all other zwave entities (sensors, switches, etc) the unique ID and key is `nodeID-valueID` or (ex: `12-3456789098`) - so we need to check 2 different things.

Reply:
I think in the future it may be better to store things in hass.data[DATA_DEVICES][nodeID][entity] so then we can just remove everything under the nodeID key